src/base-rest-resource.ts

```diff
@@ -288,25 +288,26 @@ class Base {
     });
   }
 
-  public serialize(): Body {
+  public serialize(saving = false): Body {
     const {HAS_MANY, HAS_ONE, READ_ONLY_ATTRIBUTES} = this.resource();
 
     return Object.entries(this).reduce((acc: Body, [attribute, value]) => {
       if (
-        ['session'].includes(attribute) ||
-        READ_ONLY_ATTRIBUTES.includes(attribute)
+        saving &&
+        (['session'].includes(attribute) ||
+          READ_ONLY_ATTRIBUTES.includes(attribute))
       ) {
         return acc;
       }
 
       if (attribute in HAS_MANY && value) {
         acc[attribute] = value.reduce(
           (attrAcc: Body, entry: Base) =>
-            attrAcc.concat(entry.serialize ? entry.serialize() : entry),
+            attrAcc.concat(entry.serialize ? entry.serialize(saving) : entry),
```

Comment:
We should probably also do the thing here where if `!entry.serialize`, we create a new instance so that we can `serialize` it.